File: app/src/main/java/com/androidyug/marsrover/data/model/Dataset.java

package com.androidyug.marsrover.data.model;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;


public class Dataset {

    private List<Photo> photos = new ArrayList<Photo>();
    private Map<String, Object> additionalProperties = new HashMap<String, Object>();

    /**
     * 
     * @return
     *     The photos
     */
    public List<Photo> getPhotos() {
        return photos;
    }

    /**
     * 
     * @param photos
     *     The photos
     */
    public void setPhotos(List<Photo> photos) {
        this.photos = photos;
    }

    public Map<String, Object> getAdditionalProperties() {
        return this.additionalProperties;
    }

    public void setAdditionalProperty(String name, Object value) {
        this.additionalProperties.put(name, value);
    }

}

File: app/src/main/java/com/androidyug/marsrover/data/model/Dataset.kt
package com.androidyug.marsrover.data.model

import java.util.ArrayList
import java.util.HashMap


class Dataset {

    /**
     *
     * @return
     * The photos
     */
    /**
     *
     * @param photos
     * The photos
     */
    var photos: List<Photo> = ArrayList()
    private val additionalProperties = HashMap<String, Any>()

    fun getAdditionalProperties(): Map<String, Any> {
        return this.additionalProperties
    }

    fun setAdditionalProperty(name: String, value: Any) {
        this.additionalProperties[name] = value
    }

}
